src/components/Home.tsx

```diff
@@ -0,0 +1,163 @@
+import { useQuery } from "@tanstack/react-query";
+import { RankDetails } from "../models/state";
+import { fetchTopManga, fetchTopManhwa, fetchTopManhua } from "../services/api";
+
+function Home() {
+  return (
+    <>
+      <h2 className="text-white font-semibold text-xl">Top Manga</h2>
+      <TopManga />
+      <h2 className="text-white font-semibold text-xl">Top Manhwa</h2>
+      <TopManhwa />
+      <h2 className="text-white font-semibold text-xl">Top Manhua</h2>
+      <TopManhua />
+    </>
+  );
+}
+
+export default Home;
+
+function TopManga() {
+  const { data: topMangaResults, error } = useQuery({
+    queryKey: [`fetchTopManga`],
+    queryFn: () => fetchTopManga(),
+  });
+
+  if (error) {
+    console.error("An unexpected error occurred: " + error);
+  }
+
+  if (!topMangaResults) {
+    return (
+      <div className="top-manga-container w-full h-3/10 text-center">
+        <p className="p-2 font-sans text-white">There is no manga to display</p>
+      </div>
+    );
+  }
+
+  // Filter the results to ensure that there will always be a cover image and narrow it down to the first 10 comics.
+  const topTenManga: RankDetails[] = topMangaResults
+    .filter((comic) => comic.md_covers?.[0]?.b2key)
+    .slice(0, 10);
+
+  return (
+    <div className="top-manga-container w-full h-3/10 carousel carousel-center space-x-4 bg-neutral rounded-box bg-transparent overflow-y-hidden flex align-center py-4 mb-4">
+      {topTenManga.map((mangaRec) => (
```

Comment:
You could also do `{topTenManga.map((mangaRec, index) => (...` then you can make use of the index value instead of doing `.indexOf(mangaRec)+1` 

This is just preference though.